support/metrics_plaid.py

```diff
@@ -952,95 +267,55 @@ def stability_min_running_balance(acc, txn, feedback, params):
 
-    except Exception as e:
-        score = 0
-        feedback['diversity']['error'] = str(e)
+        if metadata['checking']['investments']['earnings']:
+            bal_invest = metadata['checking']['investments']['earnings']['total_value']
 
-    finally:
-        return score, feedback
+        balance = bal_savings + bal_invest
 
+        # read params
+        w = np.digitize(acc_count, [i + 2 for i in params['count_zero']], right=False)
+        x = np.digitize(txn_timespan, params['duration'], right=True)
+        y = np.digitize(txn_mtimespan, params['due_date'], right=True)
+        z = np.digitize(balance, params['volume_invest'], right=True)
 
-# @evaluate_function
-def diversity_profile(acc, feedback, params):
-    '''
-    Description:
-        A score for number of saving and investment accounts owned
-
-    Parameters:
-        acc (list): Plaid 'Accounts' product
-        feedback (dict): score feedback
-        params (dict): model parameters, i.e. coefficients
+        # account
+        score.append(params['diversity_velo_mtx'][w][x])
 
-    Returns:
-        score (float): points scored for accounts owned
-        feedback (dict): score feedback
-    '''
+        # profile
+        score.append(params['fico_medians'][z])
 
-    try:
-        myacc = list()
-
-        acc = [
-            x
-            for x in acc
-            if x['type'] == 'loan' or int(x['balances']['current'] or 0) != 0
-        ]  # exclude $0 balance accounts
-
-        balance = 0
-        for a in acc:
-            id = a['account_id']
-            type = '{}_{}'.format(a['type'], str(a['subtype']))
-
-            # Consider savings, hda, cd, money mart, paypal, prepaid, cash management, edt accounts
-            if (type.split('_')[0] == 'depository') & (
-                type.split('_')[1] != 'checking'
-            ):
-                balance += int(a['balances']['current'] or 0)
-                myacc.append(id)
-
-            # Consider ANY type of investment account
-            if type.split('_')[0] == 'investment':
-                balance += int(a['balances']['current'] or 0)
-                myacc.append(id)
-
-        if balance != 0:
-            score = params['fico_medians'][
-                np.digitize(balance, params['volume_invest'], right=True)
-            ]
+        # update feedback
+        feedback['diversity']['bank_accounts'] = acc_count
+        feedback['stability']['loan_duedate'] = np.append(params['due_date'], 6)[y]
 
-        else:
-            raise Exception('no investing nor savings accounts')
+        if not balance:
+            raise Exception('no savings or investment accounts')
 
     except Exception as e:
-        score = 0
         feedback['diversity']['error'] = str(e)
 
     finally:
+        num, size = 2, len(score)
+        if size < num:
+            score = fill_list(score, num, size)
+        print(f'\033[36m  -> Diversity:\t{score}\033[0m')
```

Comment:
Time allowing, I'd like to walk through the refactoring changes in `metrics_plaid.py`.

Reply:
@irene-bbox have added 3 new metrics to credit model and config.json